### PR TITLE
Logout Feature + Update when agenda item updated, created or deleted

### PR DESCRIPTION
**Pull request by MKiperszmid**
Logout

### CODE REVIEW

File: app/src/main/java/com/mk/tasky/MainViewModel.kt [hunk 1]
```diff
@@ -32,4 +39,13 @@ class MainViewModel @Inject constructor(
             }
         }
     }
+
+    fun onLogout() {
+        viewModelScope.launch {
+            preferences.deleteUser()
+            withContext(defaultDispatcher) {
+                database.clearAllTables()
```
Comment (philipplackner) [inline app/src/main/java/com/mk/tasky/MainViewModel.kt:47]: Why the default dispatcher for an IO thing?

File: app/src/main/java/com/mk/tasky/agenda/home/presentation/HomeScreen.kt [hunk 2]
```diff
@@ -27,18 +29,41 @@ import java.time.format.DateTimeFormatter
...
     redirect: (HomeAgendaType, LocalDateTime) -> Unit,
+    options: (HomeItemOptions, String) -> Unit,
+    onLogout: () -> Unit,
     viewModel: HomeViewModel = hiltViewModel()
 ) {
     val state = viewModel.state
     val context = LocalContext.current
 
+    LaunchedEffect(true) {
+        viewModel.onEvent(HomeEvent.OnRefreshAgenda)
```
Comment (philipplackner) [inline app/src/main/java/com/mk/tasky/agenda/home/presentation/HomeScreen.kt:40]: Why not put this in `init` of your ViewModel?

**Review by philipplackner — APPROVED** [review]

File: app/src/main/java/com/mk/tasky/agenda/home/presentation/HomeScreen.kt (hunk 2 above)
Comment (MKiperszmid) [reply app/src/main/java/com/mk/tasky/agenda/home/presentation/HomeScreen.kt:40]: Explained over whatsapp (init doesnt get called due to backstack)

File: app/src/main/java/com/mk/tasky/MainViewModel.kt (hunk 1 above)
Comment (MKiperszmid) [reply app/src/main/java/com/mk/tasky/MainViewModel.kt:47]: Will rename variable. nice catch!

**Pull request merged**